lib/services/local/headlessauthn_watcher.go

```diff
@@ -219,33 +223,34 @@ func (h *HeadlessAuthenticationWatcher) notify(headlessAuthns ...*types.Headless
 	}
 }
 
+type WaitConditionFunc func(*types.HeadlessAuthentication) (bool, error)
```

Comment:
Because this is only used once, and we are in a general-ish package (services/local), I think it was best to declare this directly in the param, as before. Otherwise we are taking a somewhat general type name - WaitConditifionFunc - to something that's very specific to headless authn.

A rename to something more specific could do too, but naming is hard... ;)